packages/babel-types/src/definitions/core.ts

```diff
@@ -378,7 +378,7 @@ defineType("ForStatement", {
 });
 
 export const functionCommon = () => ({
-  params: validateArrayOfType("Identifier", "Pattern", "RestElement"),
+  params: validateArrayOfType("FunctionParameter"),
```

Comment:
The alias is named `FunctionParameter` in order to support the [Class Method Parameter Decorator proposal](https://github.com/estree/estree/blob/master/experimental/class-method-parameter-decorators.md) in the future, where a new `Parameter`-type AST node will be introduced.

Reply:
For Babel 9, I also want to use `Parameter` to store function params type annotations (rather than sticking them on identifier nodes).